### PR TITLE
CVSB-495

### PR DESCRIPTION
**Pull request by vladddragomir**

### CODE REVIEW

File: src/pages/testing/test-creation/test-create/test-create.html [hunk 1]
```diff
@@ -1,105 +1,101 @@
...
-            <button ion-button
-                    color="dark"
-                    (click)="onAbandonVehicleTest(vehicleTest)">Abandon
+    <ion-list class="full-dividers-list">
+        <ion-card class="card" *ngFor="let vehicle of testData.vehicles; let i = index">
+            <button ion-item detail-none class="ion-item-title"
+                    (click)="onVehicleDetails(vehicle)">
+                <ion-img item-start class="vehicle-type-icon"
+                         src="./assets/imgs/{{getVehicleTypeIconToShow(vehicle)}}-icon.svg" width="50"></ion-img>
+                <h2>{{ vehicle.vrm }} ({{vehicle.techRecord[0].vehicleType}})</h2>
```
Comment (danieliordache) [inline src/pages/testing/test-creation/test-create/test-create.html]: there you should show the tech-record with the status='current' and not techRecord[0]

File: src/pages/testing/test-creation/test-create/test-create.html [hunk 2]
```diff
@@ -1,105 +1,101 @@
...
+                            detail-none
+                            (click)="openTest(vehicle, vehicleTest)"
+                            class="bottom-divider">
+                        <h3 class="test-type">{{ vehicleTest.testTypeName }}</h3>
+                        <p item-end ion-text text-uppercase
+                           class="test-type-status test-result"
+                           [color]="commonFunctions.getTestResultColor(vehicleTest.testResult)"
+                           *ngIf="isTestAbandoned(vehicleTest)">Abandoned</p>
+                        <h3 item-end class="test-type-status"
+                            [ngClass]="{'test-type-inProgress': getTestTypeStatus(vehicleTest) !== 'Edit'}"
```
Comment (danieliordache) [inline src/pages/testing/test-creation/test-create/test-create.html]: move 'Edit' to app.enums

File: src/pages/testing/test-creation/test-review/test-review.html [hunk 3]
```diff
@@ -1,59 +1,164 @@
...
+                                       [color]="commonFunctions.getTestResultColor(testType.testResult)">
+                                        {{ testType.testResult }}
+                                    </p>
+                                </ion-col>
+                            </ion-row>
+                        </ion-grid>
+                    </ion-item>
+                    <ion-item no-padding
+                              class="bottom-divider"
+                              *ngIf="testType.numberOfSeatbeltsFitted && testType.testResult !== 'abandoned'">
```
Comment (danieliordache) [inline src/pages/testing/test-creation/test-review/test-review.html]: use the enum created for test-result

File: src/pages/testing/test-creation/test-review/test-review.html [hunk 4]
```diff
@@ -1,59 +1,164 @@
...
+                        <div margin-left class="test-field">
+                            <h3>Seatbelt installation check</h3>
+                            <h3 class="test-field-value">{{testType.seatbeltInstallationCheckDate ? 'Yes' : 'No'}}</h3>
```
Comment (danieliordache) [inline src/pages/testing/test-creation/test-review/test-review.html]: use the enum

File: src/pages/testing/test-creation/test-review/test-review.html [hunk 5]
```diff
@@ -1,59 +1,164 @@
...
+                        </div>
+                        <div margin-left class="test-field">
+                            <h3>Number of seatbelts fitted</h3>
+                            <h3 class="test-field-value">{{testType.numberOfSeatbeltsFitted}}</h3>
+                        </div>
+                        <div margin-left class="test-field">
+                            <h3>Most recent seatbelt check</h3>
+                            <h3 class="test-field-value">{{testType.lastSeatbeltInstallationCheckDate | date: 'd MMM yy'}}</h3>
```
Comment (danieliordache) [inline src/pages/testing/test-creation/test-review/test-review.html]: use/create an enum with the date format

File: src/pages/testing/test-creation/test-review/test-review.html [hunk 6]
```diff
@@ -1,59 +1,164 @@
...
+                        </div>
+                    </ion-item>
+                    <ion-item no-padding
+                              class="bottom-divider"
+                              *ngIf="testType.certificateNumber && testType.testResult !== 'abandoned'">
+                        <div margin-left class="test-field test-field-no-border-top">
+                            <h3>Certificate number</h3>
+                            <h3 class="test-field-value">{{(testType.testResult === 'pass' ? 'LP' : 'LF') + testType.certificateNumber}}</h3>
```
Comment (danieliordache) [inline src/pages/testing/test-creation/test-review/test-review.html]: use test-result enum

File: src/pages/testing/test-creation/test-review/test-review.html [hunk 7]
```diff
@@ -1,59 +1,164 @@
...
+                            <h3 class="test-field-value">{{testType.numberOfSeatbeltsFitted}}</h3>
+                        </div>
+                        <div margin-left class="test-field">
+                            <h3>Most recent seatbelt check</h3>
+                            <h3 class="test-field-value">{{testType.lastSeatbeltInstallationCheckDate | date: 'd MMM yy'}}</h3>
+                        </div>
+                    </ion-item>
+                    <ion-item no-padding
+                              class="bottom-divider"
+                              *ngIf="testType.certificateNumber && testType.testResult !== 'abandoned'">
```
Comment (danieliordache) [inline src/pages/testing/test-creation/test-review/test-review.html]: use test-result enum

File: src/pages/testing/test-creation/test-review/test-review.html [hunk 8]
```diff
@@ -1,59 +1,164 @@
...
+                        <div margin-left class="test-field test-field-no-border-top">
+                            <h3>Certificate number</h3>
+                            <h3 class="test-field-value">{{(testType.testResult === 'pass' ? 'LP' : 'LF') + testType.certificateNumber}}</h3>
+                        </div>
+                    </ion-item>
+                    <ion-item padding-left
+                              class="bottom-divider"
+                              *ngIf="testType.testResult !== 'abandoned' && getTestTypeOptionalFieldsToDisplay(testType, 'defects')">
```
Comment (danieliordache) [inline src/pages/testing/test-creation/test-review/test-review.html]: use test-result enum

File: src/pages/testing/test-creation/test-review/test-review.html [hunk 9]
```diff
@@ -1,59 +1,164 @@
...
+                                    </ion-badge>
+                                </h3>
+                                <h4 text-wrap class="parent-info-color">
+                                    {{ defect.metadata.category.imNumber }}. {{ defect.metadata.category.imDescription}}
+                                </h4>
+                                <h4 text-wrap class="parent-info-color">
+                                    {{ defect.metadata.item.itemNumber }}. {{ defect.metadata.item.itemDescription }}
+                                </h4>
+                                <h4 text-wrap class="parent-info-color"
+                                    *ngIf="defect.deficiencyCategory != 'Advisory'">
```
Comment (danieliordache) [inline src/pages/testing/test-creation/test-review/test-review.html]: use deficiency-category enum and check if it's all lowercase

File: src/pages/testing/test-creation/test-review/test-review.html [hunk 10]
```diff
@@ -1,59 +1,164 @@
...
+                                    *ngIf="defect.notes">{{defect.notes}}</h4>
+                            </div>
+                        </div>
+                        <h3 margin-bottom class="parent-info-color"
+                            *ngIf="!testType.defects.length">None
+                        </h3>
+                    </ion-item>
+                    <ion-item padding-left margin-bottom
+                              class="bottom-divider"
+                              *ngIf="testType.testResult !== 'abandoned' && getTestTypeOptionalFieldsToDisplay(testType, 'notes')">
```
Comment (danieliordache) [inline src/pages/testing/test-creation/test-review/test-review.html]: use test-result enum

File: src/pages/testing/test-creation/test-review/test-review.html [hunk 11]
```diff
@@ -1,59 +1,164 @@
...
+                        <div margin-top margin-bottom class="notes-section-title">
+                            Notes
+                        </div>
+                        <h3 text-wrap *ngIf="testType.additionalNotesRecorded.length">{{testType.additionalNotesRecorded}}</h3>
+                        <h3 class="parent-info-color" *ngIf="!testType.additionalNotesRecorded.length">None</h3>
+                    </ion-item>
+                    <ion-item padding-left margin-bottom
+                              class="bottom-divider"
+                              *ngIf="testType.testResult === 'abandoned'">
```
Comment (danieliordache) [inline src/pages/testing/test-creation/test-review/test-review.html]: use test-result enum

File: src/pages/testing/test-creation/test-review/test-review.html [hunk 12]
```diff
@@ -1,59 +1,164 @@
...
+                        <div margin-top margin-bottom class="notes-section-title">
+                            Reason(s) for abandoning
+                        </div>
+                        <h4 text-wrap class="parent-info-color"
+                            *ngFor="let reason of testType.abandonment.reasons">{{reason}}</h4>
+                    </ion-item>
+                    <ion-item padding-left margin-bottom
+                              class="bottom-divider"
+                              *ngIf="testType.testResult === 'abandoned' && testType.abandonment.additionalComment.length">
```
Comment (danieliordache) [inline src/pages/testing/test-creation/test-review/test-review.html]: use test-result enum

File: src/pages/testing/test-creation/test-review/test-review.html [hunk 13]
```diff
@@ -1,59 +1,164 @@
...
+                        <div margin-top margin-bottom class="notes-section-title">
+                            Notes
+                        </div>
+                        <h4 text-wrap class="parent-info-color">{{testType.abandonment.additionalComment}}</h4>
+                    </ion-item>
+                    <button ion-item
+                            padding-left
+                            class="bottom-divider change-details-button"
+                            *ngIf="testType.testResult !== 'abandoned'"
```
Comment (danieliordache) [inline src/pages/testing/test-creation/test-review/test-review.html]: use test-result enum

File: src/pages/testing/test-creation/test-review/test-review.scss [hunk 14]
```diff
@@ -25,6 +35,60 @@ page-test-review {
...
+          .defect-details {
+            &.border {
+              padding-bottom: 8px;
+              margin-bottom: 8px;
+              border-bottom: 0.55px solid $list-ios-border-color;
+            }
+
+            h3 {
+              white-space: normal;
+              font-size: 17px;
```
Comment (danieliordache) [inline src/pages/testing/test-creation/test-review/test-review.scss]: use/create scss variable

File: src/pages/testing/test-creation/test-create/test-create.html (hunk 2 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-create/test-create.html]: Done

File: src/pages/testing/test-creation/test-review/test-review.html (hunk 3 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-review/test-review.html]: done

File: src/pages/testing/test-creation/test-review/test-review.html (hunk 6 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-review/test-review.html]: done

File: src/pages/testing/test-creation/test-review/test-review.html (hunk 7 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-review/test-review.html]: done

File: src/pages/testing/test-creation/test-review/test-review.html (hunk 8 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-review/test-review.html]: done

File: src/pages/testing/test-creation/test-review/test-review.html (hunk 10 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-review/test-review.html]: done

File: src/pages/testing/test-creation/test-review/test-review.html (hunk 11 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-review/test-review.html]: done

File: src/pages/testing/test-creation/test-review/test-review.html (hunk 12 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-review/test-review.html]: done

File: src/pages/testing/test-creation/test-review/test-review.html (hunk 13 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-review/test-review.html]: done

File: src/pages/testing/test-creation/test-review/test-review.html (hunk 4 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-review/test-review.html]: done

File: src/pages/testing/test-creation/test-review/test-review.html (hunk 5 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-review/test-review.html]: done

File: src/pages/testing/test-creation/test-review/test-review.html (hunk 9 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-review/test-review.html]: done

File: src/pages/testing/test-creation/test-review/test-review.scss (hunk 14 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-review/test-review.scss]: done

File: src/pages/testing/test-creation/test-create/test-create.html (hunk 1 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-create/test-create.html]: done

**Pull request merged**